File: NotesServer/Models/PinboardNote.cs
using System;
using SQLite;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace NotesServer.Models
{
    public class PinboardNote : DatabaseEntity
    {
        [PrimaryKey, AutoIncrement]
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [Required]
        [JsonPropertyName("title")]
        public string Title { get; set; } = string.Empty;

        [Required]
        [JsonPropertyName("content")]
        public string Content { get; set; } = string.Empty;

        [JsonPropertyName("positionX")]
        public double PositionX { get; set; }

        [JsonPropertyName("positionY")]
        public double PositionY { get; set; }

        [JsonPropertyName("width")]
        public double Width { get; set; } = 200.0;

        [JsonPropertyName("height")]
        public double Height { get; set; } = 150.0;

        [JsonPropertyName("backgroundColor")]
        public int BackgroundColor { get; set; } = unchecked((int)0xFF424242);

        [JsonPropertyName("icon")]
        public string Icon { get; set; } = "person";
    }
} 

File: NotesServerGO/models/pinboard_note.go
package models

import "time"

// PinboardNote представляет заметку на доске.
type PinboardNote struct {
	Id              int64     `json:"id" db:"Id"`
	Title           string    `json:"title" db:"Title"`
	Content         string    `json:"content" db:"Content"`
	PositionX       float64   `json:"position_x" db:"PositionX"`
	PositionY       float64   `json:"position_y" db:"PositionY"`
	Width           float64   `json:"width" db:"Width"`
	Height          float64   `json:"height" db:"Height"`
	BackgroundColor int       `json:"background_color" db:"BackgroundColor"` // ARGB int
	IconCodePoint   int       `json:"icon" db:"IconCodePoint"`               // Храним CodePoint иконки из Flutter (map['icon'])
	DatabaseId      int64     `json:"database_id" db:"DatabaseId"`
	CreatedAt       time.Time `json:"-" db:"CreatedAt"`
	UpdatedAt       time.Time `json:"-" db:"UpdatedAt"`
}
